test/e2e/e2e_test.go

```diff
@@ -125,3 +125,265 @@ func TestClusterHealthy(t *testing.T) {
+			}, wait.WithTimeout(3*time.Minute), wait.WithInterval(10*time.Second)); err != nil {
+				t.Fatalf("Pod failed to be recreated: %v", err)
+			}
+
+			// Wait for full StatefulSet readiness
+			waitForSTSReadiness(t, c, etcdClusterName, int(initialReplicas))
+
+			// Verify PVC usage after recovery
+			verifyPodUsesPVC(t, c, targetPodName, "etcd-data-"+etcdClusterName)
+
+			// Verify member ID consistency
+			finalMembers := getEtcdMembers(t, c, podName)
+			if len(finalMembers) != initialMemberCount {
+				t.Errorf("Member count changed after recovery: expected %d, got %d", initialMemberCount, len(finalMembers))
+			}
+
+			// Verify the recovered pod maintains the same member ID
+			if targetMemberID, exists := initialMembers[targetPodName]; exists {
+				if finalMemberID, exists := finalMembers[targetPodName]; exists {
+					if targetMemberID != finalMemberID {
+						t.Errorf("Member ID changed for %s: expected %d, got %d", targetPodName, targetMemberID, finalMemberID)
+					}
+				} else {
+					t.Errorf("Member %s not found after recovery", targetPodName)
+				}
+			}
+
+			// Verify cluster replication works across recovered pod
+			_, stderr, err := execInPod(t, c, podName, namespace, []string{"etcdctl", "put", "replication-test", "value"})
+			if err != nil {
+				t.Fatalf("Failed to write data after recovery: %v, stderr: %s", err, stderr)
+			}
+
+			stdout, stderr, err := execInPod(t, c, targetPodName, namespace, []string{"etcdctl", "get", "replication-test"})
+			if err != nil {
+				t.Fatalf("Failed to read data from recovered pod: %v, stderr: %s", err, stderr)
+			}
+
+			if !strings.Contains(stdout, "value") {
+				t.Errorf("Data replication to recovered pod failed. Expected 'value', got: %s", stdout)
+			}
+
+			return ctx
+		})
+
+	feature.Teardown(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		cleanupEtcdCluster(ctx, t, c, etcdClusterName)
+		return ctx
+	})
+
+	_ = testEnv.Test(t, feature.Feature())
+}
+
+func TestEtcdClusterFunctionality(t *testing.T) {
+	feature := features.New("etcd-cluster-functionality")
+	etcdClusterName := "etcd-functionality-test"
+	testKey := "test-key"
+	testValue := "test-value"
+
+	feature.Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		createEtcdClusterWithPVC(ctx, t, c, etcdClusterName, 3)
+		waitForSTSReadiness(t, c, etcdClusterName, 3)
+		return ctx
+	})
+
+	feature.Assess("verify cluster health", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		podName := fmt.Sprintf("%s-0", etcdClusterName)
+		command := []string{"etcdctl", "endpoint", "health"}
+		stdout, stderr, err := execInPod(t, c, podName, namespace, command)
+		if err != nil {
+			t.Fatalf("Failed to check endpoint health: %v, stderr: %s", err, stderr)
+		}
+
+		if !strings.Contains(stdout, "is healthy") {
+			t.Errorf("Expected healthy endpoints, but got: %s", stdout)
+		}
```

Comment:
You only check the health of one member. Should you check all members' health?

Reply:
It's now using `--cluster` to check all members' health. I also added `waitForNoLearners`.